plutus-contract/src/Plutus/Contract/StateMachine/ThreadToken.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DerivingStrategies    #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE MonoLocalBinds        #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns        #-}
+{-# LANGUAGE NoImplicitPrelude     #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE ViewPatterns          #-}
+{-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}
+module Plutus.Contract.StateMachine.ThreadToken where
+
+import           PlutusTx.Prelude     hiding (Monoid (..), Semigroup (..))
+
+import           Data.Aeson           (FromJSON, ToJSON)
+import           GHC.Generics         (Generic)
+import           Ledger               (CurrencySymbol, TxOutRef (..))
+import qualified Ledger.Contexts      as V
+import           Ledger.Scripts
+import qualified Ledger.Typed.Scripts as Scripts
+import           Ledger.Value         (TokenName (..), Value (..))
+import qualified Ledger.Value         as Value
+import qualified PlutusTx
+import qualified PlutusTx.AssocMap    as Map
+import qualified Prelude              as Haskell
+
+data ThreadToken = ThreadToken
+    { ttOutRef         :: TxOutRef
+    , ttCurrencySymbol :: CurrencySymbol
+    }
+    deriving stock (Haskell.Eq, Haskell.Show, Generic)
+    deriving anyclass (ToJSON, FromJSON)
+
+PlutusTx.makeLift ''ThreadToken
+
+validate :: TxOutRef -> (ValidatorHash, Bool) -> V.ScriptContext -> Bool
```

Comment:
I know it's a trivial usecase, but it would make me feel less like I had to audit all the use sites if we took `data MintingPolarity = Mint | Burn` or something.